### PR TITLE
vendor(dreame): 1C voicepack install support

### PR DESCRIPTION
**Pull request by FrankZZ**
## Type of change
Type B:
- [x] New capability
- [ ] New core feature

 
# Description (Type B)

Implements custom voice pack install support for the Dreame 1C.

Feature discussion thread: maybe https://github.com/Hypfer/Valetudo/issues/165 ? I discussed this PR already with @Hypfer via Telegram.

### CODE REVIEW

File: lib/robots/dreame/capabilities/Dreame1CVoicePackManagementCapability.js [hunk 1]
```diff
@@ -0,0 +1,118 @@
...
+
+    /**
+     * This method should return the status of the current voice pack operation, if one is ongoing.
+     *
+     * @returns {Promise<ValetudoVoicePackOperationStatus>}
+     */
+    async getVoicePackOperationStatus() {
+        // Voice pack operation status is returned via events.
+        // Since the capability mechanism doesn't support events and
+        // I didn't find a get_properties way to retrieve it, it's unsupported (for now?) and will always return IDLE
```
Comment (Depau) [inline lib/robots/dreame/capabilities/Dreame1CVoicePackManagementCapability.js:107]: If this is something that is available in `onMessage()` in the robot object you could probably do something like what I did to remember the last started cleaning state operation for viomi:

https://github.com/Hypfer/Valetudo/blob/7f4df51ad1d2a80e79a7a08a5bc68f220c81177b/lib/robots/viomi/ViomiValetudoRobot.js#L40-L44

**Review by Depau — APPROVED** [review]

**Comment by Hypfer** [comment]
This is still marked as a draft. Is that intended?

**Comment by FrankZZ** [comment]
> This is still marked as a draft. Is that intended?

Haven't found the time to add current install status support as proposed by @Depau
But we could also iterate on it and add that in a next PR?

**Comment by Hypfer** [comment]
Sounds good 👍

**Comment by FrankZZ** [comment]
Woops, the close button in the GitHub app doesn't just close the PR view, it closes the PR without confirmation :)

**Pull request merged**